### DISCUSSION TITLE
Extending the simple-translation contrib packages to better support custom translation workflows

### DISCUSSION THREAD

**Discussion by NotANormalNerd** [wagtail/wagtail#11464] [Ideas]
Hello everybody,

I opened https://github.com/wagtail/wagtail/pull/11461 yesterday and would like to receive some feedback.

I opened a discussion here a few moons ago: https://github.com/wagtail/wagtail/discussions/9231 and we have been implementing some improvements for the simple translations package internally that we would really love to see in the upstream project.

The idea here is two fold: On the one hand add wagtail hooks for when a translation has been triggered from the Admin UI. This way you can easily trigger you specific translation workflow, which for most people and organizations probably is outside of Wagtail. So we added a `before_translate_page/object` and `after_translate_page/object`. (Technically this should probably be before/after_copy_for_translation_page/object)

On the other hand we implemented a parent_locale field that will be filled when we copy for translation, so we can keep track of the parent locale. We often have the situation that for example the french translation will also be copied into french-Belgium and french-Switzerland, without really changing much.

We have some more features for the simple-translation package in the pipeline, but would like to hear some feedback on those two ideas, as everything else has a dependency on this.

Thanks for your feedback.
Best regards,
Dennis Schmalacker

**Comment by zerolab** [comment]
Thank you for following up on this @NotANormalNerd. Will add a discussion point to our core meetings.

**Comment by gasman** [comment]
Discussed this in today's team meeting, and we're in favour of this - extension points such as hooks are definitely the way to go, rather than overloading the app with built-in functionality (especially for an app that's designated as 'simple').

**Reply by NotANormalNerd** [reply]
Glad to hear. I also think that Wagtail should mostly handle the "copy for translation" functionality and page management, as I already mentioned in the old discussion. So we will slowly work in that direction.

I will work on the PR and fix the Test.
